### DISCUSSION TITLE
onUpdate always receives old value of state

### DISCUSSION THREAD

**Discussion by umerjaved178** [ueberdosis/tiptap#2788] [General Questions]
Hey Fellas! Problem: onUpdate is always receiving old value of state As you can see in the attached photo that I am receiving an extra argument in debounce function i.e. notesID notesID is async, it is initially undefined and then it has value let's say after 1 sec. The problem is even though notesID will get updated in the component yet I have the old value in the onUpdate function How can I fix that issue Thanks in advance 🙂

![image](https://user-images.githubusercontent.com/25152580/168451822-af4866a1-0c7c-4ff9-a170-64da6caa4f31.png)

**Comment by codemzy** [comment]
You can pass `notesID` (and any other values the editor depends on) in the dependency array to the editor hook. e.g:

```js
const editor = new Editor({
  onUpdate({ editor }) {
    // The content has changed.
  },
}, [notesID]);
```
